Build failure with ghc-8.4.2

```
Failed to build fswatcher-0.2.1.
Build log (
/home/simon/.cabal/logs/ghc-8.4.2/fswatcher-0.2.1-d7a5f09ca4df464cda6585553e273af58d6cf6be6512f3837d8ec690da63fc2b.log
):
Configuring fswatcher-0.2.1...
Preprocessing executable 'fswatcher' for fswatcher-0.2.1..
Building executable 'fswatcher' for fswatcher-0.2.1..
[1 of 3] Compiling Opts             ( Opts.hs, dist/build/fswatcher/fswatcher-tmp/Opts.o )
[2 of 3] Compiling Pipeline         ( Pipeline.hs, dist/build/fswatcher/fswatcher-tmp/Pipeline.o )
[3 of 3] Compiling Main             ( fswatcher.hs, dist/build/fswatcher/fswatcher-tmp/Main.o )

fswatcher.hs:39:21: error:
    • The constructor ‘Modified’ should have 3 arguments, but has been given 2
    • In the pattern: Modified p _
      In an equation for ‘isThisFile’:
          isThisFile (Modified p _) = p == fromString path
      In an equation for ‘watch’:
          watch filetype m path trigger opt
            = let watchFun = ...
              in watchFun (\ _ -> void $ tryPutMVar trigger ())
            where
                isThisFile (Modified p _) = p == fromString path
                isThisFile _ = False
                matchFiles :: WatchOpt -> Event -> Bool
                matchFiles wo event
                  = let ...
                    in
                      (null includes || p =~ includePath wo)
                        && (null excludes || not (p =~ excludePath wo))
   |
39 |   where isThisFile (Modified p _) = p == fromString path
   |                     ^^^^^^^^^^^^
cabal: Failed to build fswatcher-0.2.1. See the build log above for details.
```

--- a/fswatcher.cabal
+++ b/fswatcher.cabal
@@ -20,7 +20,7 @@ Executable fswatcher
     build-depends:   base            >= 4 && < 5
                    , unix            >= 2.5
                    , process         >= 1.1
-                   , fsnotify        >= 0.1
+                   , fsnotify        >= 0.3 && < 0.4
                    , system-filepath >= 0.4
                    , directory       >= 1.2
                    , optparse-applicative >= 0.11

--- a/src/fswatcher.hs
+++ b/src/fswatcher.hs
@@ -36,8 +36,9 @@ watch filetype m path trigger opt =
                    File      -> watchDir  m (encodeString $ directory $ decodeString path) isThisFile
    in watchFun (\_ -> void $ tryPutMVar trigger ())
 
-  where isThisFile (Modified p _) = p == fromString path
-        isThisFile _              = False
+  where isThisFile :: Event -> Bool
+        isThisFile (Modified p _ _) = p == fromString path
+        isThisFile _                = False
         matchFiles :: WatchOpt -> Event -> Bool
         matchFiles wo event = let p = eventPath event
                                   includes = includePath wo

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,7 +1,8 @@
 resolver: lts-7.4
 packages:
 - '.'
-extra-deps: []
+extra-deps:
+- fsnotify-0.3.0.1
 flags: {}
 
 

